codec-http2/src/test/java/io/netty/handler/codec/http2/DataCompressionHttp2Test.java

```diff
@@ -110,7 +112,18 @@ public void teardown() throws InterruptedException {
 
     @Test
     public void justHeadersNoData() throws Exception {
-        bootstrapEnv(0);
+        // The stream will be closed upon creation. We initialize the server latch to 2 because
+        // we ensure that 1. the stream is closed, and 2. that we received a headers read.
+        bootstrapEnv(0, 2);
+        doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                // Trigger point 2 to decrement serverLatch, and so the test doesn't prematurely exit.
+                serverLatch.countDown();
```

Comment:
@nmittler  - If I add mocks to close the stream locally when I see EOS then yes. I'll commit this code and you let me know what you think.